[v6r17] Fix docstrings, markup, readthedocs tweaks

andresailer opened the pull request:
Tried to make the readthedocs build succesful again. Did not completely succeed, though it is now working at least sometimes. I think the only thing that fails is the singlehtml building, for which I now removed all of the code documentation.

Fixed almost all warnings and errors in the docstrings.


Review check list:

- [x] MakeDoc: in limited release do not create subpages
- [x] Change sphinx version to latest, keep some badly behaving files that don't completely break ....
- [x] Correct location of favicon.ico
- [x] MakeDoc: create dummy file for bad_files, add some more misbehaving things
- [x] AccountingSystem: fix docstring syntax
- [x] WorkloadManagementSystem: fix docstring syntax
- [x] Resources: fix docstring
- [x] ResourceStatusSystem: fix docstring
- [x] DataManagementSystem: fix docstring
- [x] RequestManagementSystem: fix docstring syntax
- [x] Core: fix docstring syntax param, return(s), indendation
- [x] MonitoringSystem: fix docstring param and return syntax
- [x] FrameworkSystem: docstring syntax
- [x] ConfigurationSystem: fix doc syntax, point to python:list
- [x] Interfaces: change parameter type to str or python:list where applicable
- [x] Doc: source, fix typo in path of workloadManagementSystem
- [x] TransformationSystem: Fix docstring syntax
- [x] Workflow: Fix docstring syntax
- [x] StorageManagement: Fix docstring warning
- [x] docs/fakeEnviroment: ignore more elasticsearch packages
- [x] Syntax fixes in docs/sources
- [x] Better rejection of tests, add private/special member in some cases
- [x] Fix wrong indentation, add external modules to ignore
- [x] Add to fake environment

chaen commented:
O_o Impressive piece of work ! You deserve a medal for that one !! 
Is there a way, like we have the landscape and travis, to have a validation of the documentation in a PR ?

fstagni commented:
OMG!!!

By the way, you are targeting integration branch here instead of v6r17, is it OK?

andresailer commented:
Thanks :)
Oops, changed targeted branch to v6r17. 
Better have it there or too many conflicts could emerge.

We could probably add `make html` in travis, let me see...

fstagni commented:
Now, we just need a careful review... volunteers? :)

andresailer commented:
I propose everyone picks the commit for his favourite system.

fstagni commented:
If you swear that you only touched doc and no code, I will just review OK

fstagni commented on ResourceStatusSystem/PolicySystem/StateMachine.py:
```diff
@@ -271,10 +260,7 @@ def levelOfPolicyState( self, policyResult ):
-      **policyResult** - dict
-        dictionary that must have the `Status` key.
-
+    :param doct policyResult: dictionary that must have the `Status` key.
```
`dict` not `doct`

fstagni commented on Core/Security/ProxyInfo.py:
```diff
@@ -17,24 +17,26 @@
-   'identity' : DN that generated the proxy
-   'hostname' : DIRAC host nickname
-   'VOMS'
+  :Returns: a dict with all the proxy info:
```
I think it's `:return:` not `:Returns:`

andresailer replied:
No this works: http://diracsailer.readthedocs.io/en/fixdoc/CodeDocumentation/Core/Security/ProxyInfo.html#DIRAC.Core.Security.ProxyInfo.getProxyInfo
`:Returns:`, `:returns:`, `:return:` all the same

http://www.sphinx-doc.org/en/1.4.9/domains.html#info-field-lists

fstagni commented on Core/Security/ProxyFile.py:
```diff
@@ -80,16 +80,24 @@ def deleteMultiProxy( multiProxyDict ):
+      * string -> upload file specified as proxy
+      * X509Chain -> use chain
+
+  :returns:  S_OK/S_ERROR
```
I think it's `:return:` not `:returns:`

fstagni commented on Core/Utilities/RabbitMQAdmin.py:
```diff
@@ -63,8 +64,9 @@ def deleteUser(user):
-  Returns:
-    S_OK: with a list of all users
+
+  :returns: S_OK with a list of all users
```
I think it's `:return:` not `:returns:`

fstagni commented on TransformationSystem/Agent/TransformationCleaningAgent.py:
```diff
@@ -34,9 +36,9 @@ class TransformationCleaningAgent( AgentModule ):
-  :param FileCatalogClient metadataClient: FileCatalogClient instance
+  :param ~DIRAC.DataManagementSystem.Client.DataManager.DataManager dm: DataManager instance
+  :param ~TransformationClient.TransformationClient transClient: TransformationClient instance
+  :param ~FileCatalogClient.FileCatalogClient metadataClient: FileCatalogClient instance
```
Why like this?

andresailer replied:
If it is just FileCatalogClient sphinx complains about multiple targets for it because Module and class have the same name. If I use `~FileCatalogClient.FileCatalogClient` the only target is the class. The tilde makes it only print the last part of the dotted path. DataManager is included somewhere else so the full path is given

http://diracsailer.readthedocs.io/en/fixdoc/CodeDocumentation/TransformationSystem/Agent/TransformationCleaningAgent.html

fstagni commented on docs/requirements.txt:
```diff
@@ -1,2 +1,2 @@
-Sphinx
+Sphinx==1.5.2
```
You can do `>=`

fstagni commented on RequestManagementSystem/private/RequestValidator.py:
```diff
@@ -19,32 +22,32 @@
-        if not request.Foo:
-          return S_ERROR("Foo not set")
-        return S_OK()
+            @staticmethod
```
No codeblock here?

andresailer replied:
http://diracsailer.readthedocs.io/en/fixdoc/CodeDocumentation/RequestManagementSystem/private/RequestValidator.html
It already gets syntax highlighting

fstagni commented on DataManagementSystem/DB/FileCatalogComponents/FileManagerBase.py:
```diff
@@ -1153,10 +1156,11 @@ def getFilesInDirectory( self, dirID, verbose = False, connection = False ):
-        :param allStatus : whether all replicas and file status are considered
-                          If False, take the visibleFileStatus and visibleReplicaStatus values from the configuration
+
+        :param DirID: ID of the directory
```
int

fstagni commented on DataManagementSystem/DB/FileCatalogComponents/FileManagerBase.py:
```diff
@@ -1153,10 +1156,11 @@ def getFilesInDirectory( self, dirID, verbose = False, connection = False ):
-                          If False, take the visibleFileStatus and visibleReplicaStatus values from the configuration
+
+        :param DirID: ID of the directory
+        :param path: useless
```
str

andresailer replied:
well, if the thing is not used, it doesn't really matter what type it is. How about `type unused`

fstagni replied:
Fine.

fstagni commented on DataManagementSystem/DB/FileCatalogComponents/FileManagerBase.py:
```diff
@@ -1153,10 +1156,11 @@ def getFilesInDirectory( self, dirID, verbose = False, connection = False ):
+
+        :param DirID: ID of the directory
+        :param path: useless
+        :param allStatus: whether all replicas and file status are considered
```
bool

fstagni commented on DataManagementSystem/DB/FileCatalogComponents/WithFkAndPs/FileManagerPs.py:
```diff
@@ -902,10 +902,11 @@ def getDirectoryReplicas( self, dirID, path, allStatus = False, connection = Fal
+
+        :param DirID: ID of the directory
+        :param path: useless
+        :param allStatus: whether all replicas and file status are considered
```
also here

fstagni commented on Resources/Catalog/FCConditionParser.py:
```diff
@@ -267,32 +272,38 @@ def __getConditionFromCS( self, catalogName, operationName ):
+      any catalog beforehand if you can't contact the CS...
+
+    :param catalogName: name of the catalog we want to work on
+    :param operationName: name of the operation we want to perform
```
types?

fstagni commented on Resources/Storage/FileStorage.py:
```diff
@@ -470,12 +474,13 @@ def createDirectory( self, path ):
-       :param recursive : if True, we recursively delete the subdir
+
+       :param path: single or list of path
+       :param recursive: if True, we recursively delete the subdir
```
types

fstagni commented:
Review finished...

andresailer commented:
Changed/added some more parameter types, fixed some typos.

atsareg commented:
Conflicts to be resolved. These changes are so massive that it is almost guaranteed to have conflicts. We should probably make a separate patch for them

atsareg commented:
And thanks a lot for this tremendous work !

andresailer commented:
Rebased. The only conflict was in `Core/Utilities/ElasticSearchDB.py` 
https://github.com/DIRACGrid/DIRAC/pull/3266/commits/91b640a9c91329bc7add835605d6eebf1f30df1b#diff-d7cd77a7f34b92b2559346596aec9c2fL74

Where the new parameter RESULT_SIZE was added in the docstring where I fixed the line above.

fstagni commented:
I suggest this is merged straight away before other conflicts.
